tests/test_commands.py

```diff
@@ -4712,6 +4716,34 @@ def test_psync(self, r):
         res = r2.psync(r2.client_id(), 1)
         assert b"FULLRESYNC" in res
 
+    @pytest.mark.onlynoncluster
+    def test_interrupted_command(self, r: redis.Redis):
+        """
+        Regression test for issue #1128:  An Un-handled BaseException
+        will leave the socket with un-read response to a previous
+        command.
+        """
+        print("start")
+
+        def helper():
+            try:
+                # blocking pop
+                with patch.object(
+                    socket.socket, "recv_into", side_effect=CancelledError
+                ) as mock_recv:
+                    r.brpop(["nonexist"])
+            except CancelledError:
+                print("canc")
+                pass  # we got some BaseException.
+            # if all is well, we can continue.
+            r.set("status", "down")  # should not hang
```

Comment:
Does this also exercise the "command expects int but got back str" scenario?

Reply:
No.  It excercises the originally reported scenario:  A blocking wait is initiated, and even though a new command is sent on the channel, the server is still in blocking wait mode.
It is harder to interrupt a half-read response, although not impossible by using mocking.  I can look into that.

Reply:
That scenario doesn't require half-read. It's an interrupt between send_command and parse_response:
```
send_command("EXISTS foo")
<interrupt>
send_command("GET foo")
parse_response() --> expected str, got int (from previous command)
```

Reply:
I'll add a half-read test case.